CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -63,14 +64,15 @@ public async Task<LessonDto> CreateLessonAsync(CreateLessonDto lessonDto)
             }
         }
 
-        public async Task<IList<LessonDto>> GetAllLessonsAsync()
+        public async Task<Result<IList<LessonDto>>> GetAllLessonsAsync()
         {
             var lessons = await _unitOfWork.LessonRepository.GetAllAsync();
+            var listLessons = _mapper.Map<IList<LessonDto>>(lessons);
 
-            return _mapper.Map<IList<LessonDto>>(lessons);
+            return Result<IList<LessonDto>>.GetSuccess(listLessons);
         }
 
-        public async Task<Lesson> AssignMentorToLessonAsync(AssignMentorToLessonDto ids)
+        public async Task<Result<Lesson>> AssignMentorToLessonAsync(AssignMentorToLessonDto ids)
```

Comment:
Don't forget to call .ToActionResult in controller action. Otherwise, it will change schema of response body.